leader-election-api/src/main/java/com/palantir/paxos/PaxosLearner.java

```diff
@@ -45,29 +45,44 @@
 
     /**
      * Returns learned value or null if non-exists.
+     *
+     * @deprecated use {@link #safeGetLearnedValue(long)} instead.
+     *
      */
     @Nullable
+    @Deprecated
+    default PaxosValue getLearnedValue(long seq) {
+        return safeGetLearnedValue(seq).orElse(null);
+    }
+
     @GET
     @Path("learned-value/{seq:.+}")
     @Produces(MediaType.APPLICATION_JSON)
-    PaxosValue getLearnedValue(@PathParam("seq") long seq);
+    Optional<PaxosValue> safeGetLearnedValue(@PathParam("seq") long seq);
 
     /**
      * Returns the learned value for the greatest known round or null if nothing has been learned.
+     *
+     * @deprecated use {@link #safeGetGreatestLearnedValue()} instead.
+     *
      */
     @Nullable
+    @Deprecated
+    default PaxosValue getGreatestLearnedValue() {
+        return safeGetGreatestLearnedValue().orElse(null);
+    }
+
     @GET
     @Path("greatest-learned-value")
     @Produces(MediaType.APPLICATION_JSON)
-    PaxosValue getGreatestLearnedValue();
+    Optional<PaxosValue> safeGetGreatestLearnedValue();
```

Comment:
Yeah, this is fair. As a secondary priority, might be good to apply JavaDoc on the non-deprecated methods saying what they do